adam/experiment/log_experiment.py

```diff
@@ -189,81 +215,67 @@ def learner_factory_from_params(
         raise RuntimeError("can't happen")
 
 
-def curriculum_from_params(params: Parameters):
-    curriculum_name = params.string(
-        "curriculum",
-        [
-            "m6-deniz",
-            "each-object-by-itself",
-            "pursuit",
-            "m6-preposition",
-            "m9-objects",
-            "m9-attributes",
-            "m9-relations",
-            "m9-events",
-            "m9-debug",
-            "m9-complete",
-        ],
-    )
-    if curriculum_name == "m6-deniz":
-        return (make_m6_curriculum(), [])
-    elif curriculum_name == "each-object-by-itself":
-        return (
-            # We show the learned each item 6 times,
-            # because pursuit won't lexicalize anything it hasn't seen five times.
-            list(
-                repeat(
-                    _make_each_object_by_itself_curriculum(
-                        perception_generator=GAILA_M6_PERCEPTION_GENERATOR
-                    ),
-                    10,
-                )
-            ),
-            [
-                _make_each_object_by_itself_curriculum(
-                    perception_generator=GAILA_M6_PERCEPTION_GENERATOR
-                )
-            ],
-        )
-    elif curriculum_name == "pursuit":
+def curriculum_from_params(
+    params: Parameters, language_mode: LanguageMode = LanguageMode.ENGLISH
+):
+    str_to_train_test_curriculum: Mapping[
+        str, Tuple[CURRICULUM_BUILDER, Optional[CURRICULUM_BUILDER]]
+    ] = {
+        "m6-deniz": (make_m6_curriculum, None),
+        "each-object-by-itself": (
+            build_each_object_by_itself_curriculum_train,
+            build_each_object_by_itself_curriculum_test,
+        ),
+        "pursuit": (
+            build_pursuit_curriculum,
+            build_each_object_by_itself_curriculum_test,
+        ),
```

Comment:
@gabbard Mypy is telling me this entry doesn't match the typing information provided but I can't spot the difference?

Reply:
Could it be due to this argument for `build_pursuit_curriculum`?    

```
pursuit_curriculum_params: Parameters,
```